pyirf/irf/energy_dispersion.py

```diff
@@ -152,12 +128,33 @@ def energy_dispersion_to_migration(
     for idx, e_true in enumerate(
             (new_true_energy_edges[1:] + new_true_energy_edges[:-1])/2
     ):
+
+        # get migration for the new true energy bin
         e_true_dispersion = true_energy_interpolation(e_true)
 
-        migration_matrix[idx, :, :] = _interp_dispersion(
+        # for details of the interpolation see
+        # https://github.com/gammapy/gammapy/blob/f12ae6acece0ff5a6f269ca4f409f38c0714e23a/gammapy/irf/energy_dispersion.py#L242
+
+        # interpolate the cumsum of the dispersion values along the migration axis
+        cumsum = np.cumsum(e_true_dispersion, axis=0)
+        with np.errstate(invalid="ignore"):
+            cumsum = np.nan_to_num(cumsum / cumsum[-1])
+        f = interpolate.interp1d(
```

Comment:
Using `np.interp` should do the same and is faster:

```
In [7]: %timeit np.interp(x, xp, yp)
3.47 µs ± 10.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [8]: %%timeit
   ...: f = interp1d(xp, yp, kind='linear')
   ...: f(x)
   ...: 
   ...: 
45.9 µs ± 215 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
What values did you interpolate there?
For me `np.interp` fails due to the additional `fov_offset` dimension.

Reply:
The advantage of `interp1d` is the additional `axis` argument. You can interpolate along a given axis only and leave the other untouched. So you could replace the loop over true energy here...

Reply:
Ah, I just used 1d. if scipy directly supports 2d, very good reason to use scipy

Reply:
About the true energy loop:
Isn't this needed, because the target binning changes for each true energy?